### PR TITLE
Refined reaction and sitting animations

### PR DESCRIPTION
**Pull request by hyperlogic**
Refinements to reaction animations and sitting animations.  Including:

* Seated state now allows for talking animations to play while seated. (Assuming: they are using the correct version of the sit script, See PR #15974)
* Seated state now allows for reactions to play while seated. However the animations are not present, the sitting idle is used as a place holder. (Assuming: they are using the correct version of the sit script  See PR #15974).
* Reaction animations now have several variations.
* Addition of "EvaluateBoth" interpType to AnimStateMachine and AnimRandomSwitch nodes.  Instead of taking a snapshot of one or two of the interpolation targets, it will continue to evaluate both branches while blending between the two over the given interpDuration.  This was added to help smooth the transition into and out of the reactions.
* Addition of a easingType parameter to AnimStateMachine and AnimRandomSwitch nodes.  This gives access to several easing functions to better control the interpolation between two states.  See the easing functions documented here: https://easings.net/en.
* Split getReactions() API into two calls:
    * getTriggerReactions() - lists all reactions that can be triggered with MyAvatar.triggerReaction().
    * getBeginReactions() - lists all reactions that can be used with MyAvatar.beginReaction() and MyAvatar.endReaction().

https://highfidelity.atlassian.net/browse/DEV-328

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20600/HighFidelity-Beta-PR15992-01a662d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20600/HighFidelity-Beta-Interface-PR15992-01a662d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20600/HighFidelity-Beta-PR15992-01a662d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20600/HighFidelity-Beta-Interface-PR15992-01a662d.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20600/HighFidelity-Beta-PR15992-01a662d.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20600/HighFidelity-Quest-Beta-PR15992-01a662d.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20630/HighFidelity-Beta-PR15992-6e64402.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20630/HighFidelity-Beta-Interface-PR15992-6e64402.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20630/HighFidelity-Beta-PR15992-6e64402.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20630/HighFidelity-Beta-Interface-PR15992-6e64402.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20630/HighFidelity-Beta-PR15992-6e64402.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20630/HighFidelity-Quest-Beta-PR15992-6e64402.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20657/HighFidelity-Beta-PR15992-4f5d0d1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20657/HighFidelity-Beta-Interface-PR15992-4f5d0d1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20657/HighFidelity-Beta-PR15992-4f5d0d1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20657/HighFidelity-Quest-Beta-PR15992-4f5d0d1.apk)

**Comment by AndrewMeadows** [comment]
Jenkins, rebuild this please.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20662/HighFidelity-Beta-PR15992-4f5d0d1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20662/HighFidelity-Quest-Beta-PR15992-4f5d0d1.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20662/HighFidelity-Beta-PR15992-4f5d0d1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20662/HighFidelity-Beta-Interface-PR15992-4f5d0d1.dmg

**Comment by hyperlogic** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20677/HighFidelity-Beta-PR15992-4f5d0d1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20677/HighFidelity-Beta-Interface-PR15992-4f5d0d1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20677/HighFidelity-Beta-PR15992-4f5d0d1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20677/HighFidelity-Quest-Beta-PR15992-4f5d0d1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20677/HighFidelity-Beta-PR15992-4f5d0d1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20677/HighFidelity-Beta-Interface-PR15992-4f5d0d1.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20683/HighFidelity-Beta-PR15992-ee4bb23.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20683/HighFidelity-Quest-Beta-PR15992-ee4bb23.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20683/HighFidelity-Beta-PR15992-ee4bb23.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20683/HighFidelity-Beta-Interface-PR15992-ee4bb23.dmg

**Comment by hyperlogic** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20693/HighFidelity-Beta-PR15992-ee4bb23.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20693/HighFidelity-Beta-Interface-PR15992-ee4bb23.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20693/HighFidelity-Beta-PR15992-ee4bb23.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20693/HighFidelity-Beta-Interface-PR15992-ee4bb23.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20693/HighFidelity-Beta-PR15992-ee4bb23.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20693/HighFidelity-Quest-Beta-PR15992-ee4bb23.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20702/HighFidelity-Beta-PR15992-727c047.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20702/HighFidelity-Quest-Beta-PR15992-727c047.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20702/HighFidelity-Beta-PR15992-727c047.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20702/HighFidelity-Beta-Interface-PR15992-727c047.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20702/HighFidelity-Beta-PR15992-727c047.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20702/HighFidelity-Beta-Interface-PR15992-727c047.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20711/HighFidelity-Beta-PR15992-f843c01.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20711/HighFidelity-Beta-Interface-PR15992-f843c01.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20711/HighFidelity-Beta-PR15992-f843c01.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20711/HighFidelity-Beta-Interface-PR15992-f843c01.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20711/HighFidelity-Beta-PR15992-f843c01.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20711/HighFidelity-Quest-Beta-PR15992-f843c01.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20738/HighFidelity-Beta-PR15992-31762a0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20738/HighFidelity-Quest-Beta-PR15992-31762a0.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20738/HighFidelity-Beta-PR15992-31762a0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20738/HighFidelity-Beta-Interface-PR15992-31762a0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20738/HighFidelity-Beta-PR15992-31762a0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20738/HighFidelity-Beta-Interface-PR15992-31762a0.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20749/HighFidelity-Beta-PR15992-a8d7f06.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20749/HighFidelity-Beta-Interface-PR15992-a8d7f06.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20749/HighFidelity-Beta-PR15992-a8d7f06.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20749/HighFidelity-Beta-Interface-PR15992-a8d7f06.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20749/HighFidelity-Beta-PR15992-a8d7f06.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20749/HighFidelity-Quest-Beta-PR15992-a8d7f06.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20751/HighFidelity-Beta-PR15992-649ef6c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20751/HighFidelity-Beta-Interface-PR15992-649ef6c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20751/HighFidelity-Beta-PR15992-649ef6c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20751/HighFidelity-Quest-Beta-PR15992-649ef6c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20751/HighFidelity-Beta-PR15992-649ef6c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20751/HighFidelity-Beta-Interface-PR15992-649ef6c.exe

File: libraries/animation/src/AnimUtil.cpp [hunk 1]
```diff
@@ -211,3 +211,95 @@ bool findPointKDopDisplacement(const glm::vec3& point, const AnimPose& shapePose
...
+    case EasingType_EaseInQuad:
+        return alpha * alpha;
+    case EasingType_EaseOutQuad:
+        return -(alpha * (alpha - 2.0f));
+    case EasingType_EaseInOutQuad:
+        return (alpha < 0.5f) ? (2.0f * alpha * alpha) : ((-2.0f * alpha * alpha) + (4.0f * alpha) - 1.0f);
+    case EasingType_EaseInCubic:
+        return alpha * alpha * alpha;
+    case EasingType_EaseOutCubic:
+        {
```
Comment (luiscuenca) [inline libraries/animation/src/AnimUtil.cpp]: > In the event that brackets are required to create local scope, the open bracket should appear on the same line as the case

File: libraries/animation/src/AnimUtil.cpp [hunk 2]
```diff
@@ -211,3 +211,95 @@ bool findPointKDopDisplacement(const glm::vec3& point, const AnimPose& shapePose
...
+            float temp = alpha - 1.0f;
+            return temp * temp * temp + 1.0f;
+        }
```
Comment (luiscuenca) [inline libraries/animation/src/AnimUtil.cpp]: We should add a break here, or remove all breaks after the function returns.

File: libraries/animation/src/AnimUtil.cpp (hunk 1 above)
Comment (luiscuenca) [reply libraries/animation/src/AnimUtil.cpp]: Same problem on lines 250 and 266

File: libraries/animation/src/AnimUtil.cpp (hunk 2 above)
Comment (luiscuenca) [reply libraries/animation/src/AnimUtil.cpp]: Same problem in line 269

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20758/HighFidelity-Beta-PR15992-640c058.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20758/HighFidelity-Beta-Interface-PR15992-640c058.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20758/HighFidelity-Beta-PR15992-640c058.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20758/HighFidelity-Quest-Beta-PR15992-640c058.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20758/HighFidelity-Beta-PR15992-640c058.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20758/HighFidelity-Beta-Interface-PR15992-640c058.exe

**Pull request merged**